inc/namespace.php

```diff
@@ -103,3 +105,23 @@ function set_vary_headers() {
 	// Set the Vary headers.
 	header( 'Vary: ' . implode( ', ', $supported_vary_headers ), false );
 }
+
+/**
+ * Add header key and custom data.
+ *
+ * @param array $key Key for the header, or array of keys.
+ * @param array $data Data to pass to the HeaderData class.
+ *
+ * @return array The header data.
+ */
+function add_header_data( array $key = null, array $data = null ) : array {
```

Comment:
@jazzsequence Curious of your thoughts about this function name. Struggled with naming and open to any suggestions.

Reply:
This adds support for new vary headers, yeah? Maybe `update_vary_headers`?

Reply:
Yeah, that sounds better IMO.